go/vt/mysqlctl/replication.go

```diff
@@ -228,6 +228,27 @@ func (mysqld *Mysqld) GetServerUUID(ctx context.Context) (string, error) {
 	return conn.Conn.GetServerUUID()
 }
 
+// GetServerStatus returns the server statuses asked for.
+func (mysqld *Mysqld) GetServerStatus(ctx context.Context, statuses []string) ([]string, error) {
+	query := `SELECT variable_value FROM performance_schema.global_status WHERE variable_name IN (`
+	for _, status := range statuses {
+		query += `"` + status + `"`
+	}
+	query += ");"
+	qr, err := mysqld.FetchSuperQuery(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	if len(qr.Rows) != len(statuses) {
```

Comment:
If this is external facing, we should identify the unknown variables. We can get `select variable_name, variable_value` to find out which variables have no values.

Also is it possible that there is no row for a valid variable because it was not enabled in the server? 

I have started using the `qr.Named().Rows`  approach since that allows named columns rather than hard coding an index. Protects against panics in addition of being self-documenting.

Reply:
Fixed